File: Java/DesignPatterns/src/main/java/designPatterns/structural/proxy/Proxy.java
package designPatterns.structural.proxy;

public class Proxy implements Subject {
  private RealSubject realSubject;
  private String value;

  public Proxy(String value) {
    this.value = value;
  }

  @Override
  public String display() {
    if (realSubject == null) {
      realSubject = new RealSubject(value);
    }
    return realSubject.display();
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/structural/proxy/Proxy.kt
package structural.proxy

data class Proxy(private var value: String) : Subject {
  private var realSubject: RealSubject? = null

  override fun display(): String {
    if (realSubject == null) {
      realSubject = RealSubject(value)
    }
    return realSubject!!.display()
  }
}